native/core/src/util.rs

```diff
@@ -21,3 +21,8 @@ pub(crate) fn is_zero(num: &u32) -> bool {
 pub(crate) fn is_false(val: &bool) -> bool {
     !(*val)
 }
+
+pub fn do_vecs_match<T: PartialEq>(a: &[T], b: &[T]) -> bool {
+    let matching = a.iter().zip(b.iter()).filter(|&(a, b)| a == b).count();
+    matching == a.len() && matching == b.len()
+}
```

Comment:
why not do `a == b`? This seems surprisingly complicated and unnecessarily inefficient.

Reply:
This match should be successful even if some elements are duplicated or the order of elements is all messed up.